internal/log/log.go

```diff
@@ -59,96 +56,111 @@ type CtxKeyAccount struct{}
+	handlerOptions := getHandlerOptions(getLevelFromVerbose(verbose))
+	handlers := []slog.Handler{getHandler(os.Stderr, handlerOptions)}
 
-// WithCtxFields creates a logger instance with preset structured logging [field.Field] based on the Context
-// Coordinate (via [CtxKeyCoord]) and environment (via [CtxKeyEnv] [CtxValEnv]) information is added to logs from the Context
-func WithCtxFields(ctx context.Context) loggers.Logger {
-	loggr := std
-	f := make([]field.Field, 0, 2)
-	if c, ok := ctx.Value(CtxKeyCoord{}).(coordinate.Coordinate); ok {
-		f = append(f, field.Coordinate(c))
-	}
-	if e, ok := ctx.Value(CtxKeyEnv{}).(CtxValEnv); ok {
-		f = append(f, field.Environment(e.Name, e.Group))
+	if loggerSpy != nil {
+		handlers = append(handlers, getHandler(loggerSpy, handlerOptions))
 	}
 
-	if a := ctx.Value(CtxKeyAccount{}); a != nil {
-		f = append(f, field.F("account", a))
+	if fileLogging && fs != nil {
+		logFile, errorFile, err := PrepareLogFiles(ctx, fs, enableMemstatLogging)
+		if err != nil {
+			Warn("Error preparing log files: %s", err.Error())
+		}
+
+		if logFile != nil {
+			handlers = append(handlers, getHandler(logFile, handlerOptions))
+		}
+
+		if errorFile != nil {
+			handlers = append(handlers, getHandler(errorFile, getHandlerOptions(slog.LevelError)))
+		}
 	}
 
-	if c, ok := ctx.Value(CtxGraphComponentId{}).(CtxValGraphComponentId); ok {
-		f = append(f, field.F("gid", c))
+	if len(handlers) == 1 {
+		return handlers[0]
 	}
-	return loggr.WithFields(f...)
-}
 
-var (
-	std loggers.Logger = console.Instance
-)
+	return NewTeeHandler(handlers...)
+}
 
-func PrepareLogging(ctx context.Context, fs afero.Fs, verbose bool, loggerSpy io.Writer, fileLogging bool, enableMemstatLogging bool) {
-	loglevel := loggers.LevelInfo
+func getLevelFromVerbose(verbose bool) slog.Level {
 	if verbose {
-		loglevel = loggers.LevelDebug
+		return slog.LevelDebug
 	}
 
-	var logFile, errFile afero.File
-	var err error
-	if fileLogging && fs != nil {
-		logFile, errFile, err = prepareLogFiles(ctx, fs, enableMemstatLogging)
+	return slog.LevelInfo
+}
+
+func getHandlerOptions(level slog.Leveler) *slog.HandlerOptions {
+	return &slog.HandlerOptions{
+		Level:       level,
+		ReplaceAttr: getReplaceAttrFunc(),
+	}
+}
+
+func getHandler(w io.Writer, options *slog.HandlerOptions) slog.Handler {
+	if shouldUseJSON() {
+		return slog.NewJSONHandler(w, options)
 	}
 
-	logFormat := loggers.ParseLogFormat(os.Getenv(loggers.EnvVarLogFormat))
-	logTime := loggers.ParseLogTimeMode(os.Getenv(loggers.EnvVarLogTime))
+	return slog.NewTextHandler(w, options)
+}
 
-	setDefaultLogger(loggers.LogOptions{
-		File:        logFile,
-		ErrorFile:   errFile,
-		JSONLogging: logFormat == loggers.LogFormatJSON,
-		LogLevel:    loglevel,
-		LogSpy:      loggerSpy,
-		LogTimeMode: logTime,
-	})
+func getReplaceAttrFunc() func(groups []string, a slog.Attr) slog.Attr {
```

Comment:
Should we maybe structure this a bit differently? Instead of deciding inside this function for every attribute if UTC should be used, we set a different replace function
```go
func getHandlerOptions(level slog.Leveler) *slog.HandlerOptions {
  if useUTC {
    return &slog.HandlerOptions{
		Level:       level,
		ReplaceAttr: getReplaceUTC,
      }
  }
return &slog.HandlerOptions{
  Level:       level,
  }
}

func getReplaceUTC() func(groups []string, a slog.Attr) slog.Attr {
		if a.Key == slog.TimeKey {
			t := a.Value.Time()
			return slog.Attr{Key: slog.TimeKey, Value: slog.TimeValue(t.UTC())}
		}

		return a
}
```
or we decide in a generic `getReplaceAttrFunc` which function to return